src/Acl.php

```diff
@@ -675,21 +674,17 @@ public function setRule(
     /**
      * Returns all child resources from the given resource.
      *
-     * @return ResourceInterface[]
+     * @param (ResourceInterface|null)[] $return
+     * @return (ResourceInterface|null)[]
      */
-    protected function getChildResources(ResourceInterface $resource)
```

Comment:
Uhh, we should really not mess with `protected` methods 🤔

They are very BC prone (mistake in the original design of the library, which is very old), so we should refrain from changing behaviour.

It would be best to make a new method instead, and deprecating the old one 🤔